shell/tools/detect-unused-files.ts

```diff
@@ -0,0 +1,111 @@
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+/* eslint-disable no-console */
+import findUnusedModule from './find-unused-module';
+import path from 'path';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const pkg = require(path.resolve(__dirname, '../package.json'));
+const { devDependencies, dependencies } = pkg;
+const nodeModules = Object.keys(dependencies)
+  .concat(Object.keys(devDependencies))
+  .concat(['path', 'prop-types', 'history']);
+
+const resolve = (pathname: string) => path.resolve(__dirname, '..', pathname);
+
+const aliasMap = {
+  app: resolve('./app'),
+  common: resolve('./app/common'),
+  configForm: resolve('./app/configForm'),
+  'yml-chart': resolve('./app/yml-chart'),
+  'config-page': resolve('./app/config-page'),
+  layout: resolve('./app/layout'),
+  user: resolve('./app/user'),
+  charts: resolve('./app/charts'),
+  dcos: resolve('./app/modules/dcos'),
+  project: resolve('./app/modules/project'),
+  publisher: resolve('./app/modules/publisher'),
+  cmp: resolve('./app/modules/cmp'),
+  org: resolve('./app/modules/org'),
+  application: resolve('./app/modules/application'),
+  runtime: resolve('./app/modules/runtime'),
+  dop: resolve('./app/modules/dop'),
+  addonPlatform: resolve('./app/modules/addonPlatform'),
+  msp: resolve('./app/modules/msp'),
+  apiManagePlatform: resolve('./app/modules/apiManagePlatform'),
+  agent: resolve('./app/agent.js'),
+  i18n: resolve('./app/i18n.ts'),
+  'dice-env': resolve('./app/external/env.ts'),
+  'monitor-overview': resolve('./app/modules/msp/monitor/monitor-overview'),
+  'application-insight': resolve('./app/modules/msp/monitor/application-insight'),
+  'external-insight': resolve('./app/modules/msp/monitor/external-insight'),
+  'service-insight': resolve('./app/modules/msp/monitor/service-insight'),
+  'browser-insight': resolve('./app/modules/msp/monitor/browser-insight'),
+  'gateway-ingress': resolve('./app/modules/msp/monitor/gateway-ingress'),
+  'docker-container': resolve('./app/modules/msp/monitor/docker-container'),
+  'mobile-insight': resolve('./app/modules/msp/monitor/mobile-insight'),
+  'api-insight': resolve('./app/modules/msp/monitor/api-insight'),
+  'trace-insight': resolve('./app/modules/msp/monitor/trace-insight'),
+  'monitor-common': resolve('./app/modules/msp/monitor/monitor-common'),
+  topology: resolve('./app/modules/msp/monitor/topology'),
+  'status-insight': resolve('./app/modules/msp/monitor/status-insight'),
+  'error-insight': resolve('./app/modules/msp/monitor/error-insight'),
+  'monitor-alarm': resolve('./app/modules/msp/monitor/monitor-alarm'),
```

Comment:
It's better to define the map in webpack.config.js and import to here to keep it sync in two places.